### ISSUE TITLE
Update autodl-irssi plugin

### ISSUE THREAD

**Issue by smoresmores** [QuickBox/QB#114]
I've recently checked and saw that the autodl-irssi plugin version is v1.62 (from March 2016).
According to their official github the newest version is v2.1.0 from 3 days ago. Not sure exactly what are the differences between the regular version and the community version.
https://github.com/autodl-community/autodl-rutorrent/releases
https://github.com/autodl-community/autodl-trackers/releases

I tried to run the update from the plugin itself but got an error:
`Undefined subroutine &AutodlIrssi::updateFailed`

Would be great if this can be update.
Thanks for your hard work!

**Comment by JMSDOnline** [comment]
I am updating this tonight

**Comment by smoresmores** [comment]
Thanks. Although when I restarted rtorrent, it now doesn't load. :(

**Comment by kade-robertson** [comment]
`[28.01.2018 18:09:35] Error downloading files. Make sure autodl-irssi is started and configured properly (eg. password, port number): Error getting files listing: Missing PHP modules: xml`

This only started happening quite recently. Related at all?

**Comment by JMSDOnline** [comment]
@smoresmores, it looks like it is looking for an old libcrypto library. Perhaps you recently did an `apt-get update` and it updated, thus rtorrent is spitting errors. If you're running 0.9.6 (current, I assume you are) then you could run `apt -y install rtorrent`. This should fix this issue as I have encountered it before on a couple of installs. This all started happening after openssl did an upstream push several updates ago.

@kade-robertson, I have had a couple of reports with this, I am going to have a look and see if QuickBox gave a bad update somewhere.. or perhaps didn't include something along the way.

**Comment by smoresmores** [comment]
@JMSDOnline yes, this solved it. It was looking for a **newer** version of libcrypto (1.0.2 instead of 1.0.0). Anyway, now everything works fine. Thanks you!!

**Comment by JMSDOnline** [comment]
> It was looking for a **newer** version
Ha, that's the one I meant. Glad that got it for you!

> Thanks you!!
Oh no my friend, thanks you!!! 😃 

Is the autodl-irssi plugin working out ok, or have you received the :xml: error as well?

**Comment by JMSDOnline** [comment]
Possible errors may exist, however @kade-robertson has worked out a fix for the initial install of the autodl-irssi package here #118

**Comment by JMSDOnline** [comment]
As per this Community topic, [this](https://plaza.quickbox.io/t/cant-update-autodl-irssi/4458/25) seems to now be resolved.

**Comment by JMSDOnline** [comment]
@kade-robertson, noted: I'll see how this can be reproduced. Could you tell me what the exact error was once more?

**Comment by kade-robertson** [comment]
Sure:

`[15.03.2018 09:58:54] Error downloading files. Make sure autodl-irssi is started and configured properly (eg. password, port number): Error getting files listing: Missing PHP modules: xml`

**Comment by JMSDOnline** [comment]
Does any of this help?

```
apt-get -y install php7.0-xml
phpenmod -v 7.0 xml
```

**Comment by kade-robertson** [comment]
`apt-get -y install php7.0-xml`:
```
Reading package lists... Done
Building dependency tree
Reading state information... Done
php7.0-xml is already the newest version (7.0.28-1+ubuntu16.04.1+deb.sury.org+1).
0 upgraded, 0 newly installed, 0 to remove and 10 not upgraded.
```

`phpenmod -v 7.0 xml`:
```
<no output>
```

I did notice that my `php -v` shows I'm using `php 7.2.3`, so I re-did both commands for `php7.2-xml` instead and ended up with the same output essentially. No changes to the current situation.

**Comment by JMSDOnline** [comment]
> I did notice that my `php -v` shows I'm using `php 7.2.3`

Hmm, that could be a problem. Your php version should be as follows:
```
php -v
PHP 7.0.28-1+ubuntu16.04.1+deb.sury.org+1 (cli) (built: Mar  6 2018 10:44:15) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.0.28-1+ubuntu16.04.1+deb.sury.org+1, Copyright (c) 1999-2017, by Zend Technologies
```

I haven't tested with anything higher than this version. 

You may have to edit your `php.ini` file and send a restart to php-fpm as per these edits:
[quickbox-setup Lines 924-941](https://github.com/QuickBox/QB/blob/master/setup/quickbox-setup#L924-#L941)

**Comment by kade-robertson** [comment]
A-ha! I think I've figured out how to fix this (albeit I'm not sure which change did it since I did them all at once), while keeping all of the existing `php` versions. Substitute any references to 7.2 with whatever the current version showing up with `php -v` is.

1. Run those two `sed` commands linked to above.
2. `a2dismod php7.2`
3. `a2enmod php7.0`
4. `update-alternatives --config php`, pick whichever index has `php7.0` and press Enter
5. `service apache2 restart`
